source/binary/index.rst

```diff
@@ -0,0 +1,287 @@
+	
+Accès
+~~~~~
+
+Accède la première valeur à la clé 0 donc b'exemple' dans l'exemple ci-dessus.
+
+.. code-block:: python
+
+	msg[0]
+	
+Opérations
+~~~~~~~~~~
+
+En plus des opérations ci-dessous voir les opérations de bytes ci-dessus.
+
+.. code-block:: python
+	# Cast bytes à bytearray
+	mutable_bytes = bytearray(b'\x00\x0F')
+
+	# Cast bytearray à bytes
+	immutable_bytes = bytes(mutable_bytes)
+
+	
+MemoryView
+----------
+
+Une memoryView est un objet qui sert d'API pour utiliser des buffers (C objet) afin de pouvoir le manipuler comme tout autre objet.
+
+Instanciation
+~~~~~~~~~~~~~
+
+.. code-block:: python 
+
+	#Crée une memoryview à partir de l'objet qui définit le nouveau buffer.
+	PyObject *PyMemoryView_FromObject(PyObject *obj)
+
+	#Crée une memoryview et wrappe le buffer en structure view.
+	#La memoryview détient le buffer qui sera désalloué automatiquement lors de la destruction de l'objet.
+	PyObject *PyMemoryView_FromBuffer(Py_buffer *view)
+
+	#Crée une memoryview d'une partie mémoire contiguë.
+	#Si dans la mémoire l'objet est stocké de manière contiguë, le pointeur pointe sur cette zone mémoire
+	#sinon une copie est faite.
+	PyObject *PyMemoryView_GetContiguous(PyObject *obj, int buffertype, char order)
+
+Opérations
+~~~~~~~~~~
+
+.. code-block:: python 
+
+	#Retourne true si l'objet obj est une memoryview.
+	int PyMemoryView_Check(PyObject *obj)
+	
+	#Retourne un pointer sur buffer wrapper par l'objet donné.
+	Py_buffer *PyMemoryView_GET_BUFFER(PyObject *obj)
+	
+Exemple d'utilisation
+~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python 
+
+	mybuf = ... # un grand buffer de bytes
+	mv_mybuf = memoryview(mybuf) # une memoryview de mybuf
+	func(mv_mybuf[:len(mv_mybuf)//2])
+	# passe la première moitié de mybuf dans func comme une "sous-view" crée par le découpage de la memoryview
+	# Aucune copie n'est faite ici!
+
+Avec bytearray:
+
+.. code-block:: python 
+
+	>>> buf = bytearray(b'abcdefgh')
+	>>> mv = memoryview(buf)
+	>>> mv[4:6] = b'ZA'
+	>>> buf
+	bytearray(b'abcdZAgh')
+
+	
+Struct
+------
+
+struct permet de convertir des structures C en valeurs de python représenté sous forme d'objets bytes.
+
+Instanciation
+~~~~~~~~~~~~~
+
+.. code-block:: python 
+
+	#Crée une memoryview à partir de l'objet qui définit le nouveau buffer.
+	PyObject *PyMemoryView_FromObject(PyObject *obj)
+
+	#Crée une memoryview et wrappe le buffer en structure view.
+	#La memoryview détient le buffer et il sera désalloué automatiquement lors de la destruction de l'objet.
+	PyObject *PyMemoryView_FromBuffer(Py_buffer *view)
+
+	#Crée une memoryview d'une partie mémoire contiguë.
+	#Si dans la mémoire l'objet est stocké de manière contiguë, le pointeur pointe sur cette 
+	#zone mémoire sinon une copie est faite.
+	PyObject *PyMemoryView_GetContiguous(PyObject *obj, int buffertype, char order)
+
+Opérations
+~~~~~~~~~~
+
+.. image:: ./img/struct_operation.jpg
+   :align: right
+   :alt: operations struct
```

Comment:
La capture d'écran d'un document HTML c'est pas terrible. Faite un lien.

1. https://docs.python.org/3/library/struct.html?highlight=struct#functions-and-exceptions
2. Show Source
3. .???.
4. _profit!_